tests/integration/api_container_test.py

```diff
@@ -880,6 +881,30 @@ def test_logs_streaming_and_follow(self):
 
         assert logs == (snippet + '\n').encode(encoding='ascii')
 
+    def test_logs_streaming_and_follow_and_cancel(self):
+        snippet = 'Flowering Nights (Sakuya Iyazoi)'
+        container = self.client.create_container(
+            BUSYBOX, 'sh -c "echo \\"{0}\\" && sleep 3"'.format(snippet)
+        )
+        id = container['Id']
+        self.tmp_containers.append(id)
+        self.client.start(id)
+        logs = six.binary_type()
+
+        generator = self.client.logs(id, stream=True, follow=True)
+
+        exit_timer = threading.Timer(3, os._exit, args=[1])
```

Comment:
To avoid this, we could bring in the `pytest-timeout` module, if that sounds better?